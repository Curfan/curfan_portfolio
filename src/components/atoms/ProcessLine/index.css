.process-line {
	& > .dash {
		display: flex;
		width: 4px;
		height: 1px;
		background: white;
		transition: background 0.3s ease-in;

		&:not(:last-child) {
			margin-bottom: 12px;
		}
	}
}
