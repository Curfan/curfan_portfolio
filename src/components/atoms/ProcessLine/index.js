import React from 'react';
import classnames from 'classnames';

import styles from './index.css';

const renderDash = count => {
	const nodes = [];

	for (let i = 1; i <= count; i += 1) {
		nodes.push(<span key={i} className={classnames(styles.dash)} />);
	}

	return nodes;
};

const ProcessLine = ({ className }) => (
	<div className={classnames(styles.processLine, className)}>{renderDash(5)}</div>
);

export default ProcessLine;
