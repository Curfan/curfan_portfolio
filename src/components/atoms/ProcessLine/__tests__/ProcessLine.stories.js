import React from 'react';
import { storiesOf } from '@storybook/react';

import ProcessLine from 'components/atoms/ProcessLine';

const stories = storiesOf('atoms/ProcessLine', module);

stories.add('__interactive', () => <ProcessLine />);
