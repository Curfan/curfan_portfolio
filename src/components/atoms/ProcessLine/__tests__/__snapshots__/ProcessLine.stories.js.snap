// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="processLine"
  >
    <span
      class="dash"
    />
    <span
      class="dash"
    />
    <span
      class="dash"
    />
    <span
      class="dash"
    />
    <span
      class="dash"
    />
  </div>
</DocumentFragment>
`;
